File: Backend/api-gateway/src/proto/transaction/Transaction.ts
// Original file: src/proto/transaction.proto

import type { TransactionStatus as _transaction_TransactionStatus, TransactionStatus__Output as _transaction_TransactionStatus__Output } from '../transaction/TransactionStatus';
import type { TransactionType as _transaction_TransactionType, TransactionType__Output as _transaction_TransactionType__Output } from '../transaction/TransactionType';

export interface Transaction {
  'id'?: (string);
  'senderId'?: (string);
  'senderAccountId'?: (string);
  'senderAccountNumber'?: (string);
  'receiverId'?: (string);
  'receiverAccountId'?: (string);
  'receiverAccountNumber'?: (string);
  'amount'?: (number | string);
  'currency'?: (string);
  'status'?: (_transaction_TransactionStatus);
  'note'?: (string);
  'paymentCode'?: (string);
  'model'?: (string);
  'callNumber'?: (string);
  'createdAt'?: (string);
  'type'?: (_transaction_TransactionType);
}

export interface Transaction__Output {
  'id': (string);
  'senderId': (string);
  'senderAccountId': (string);
  'senderAccountNumber': (string);
  'receiverId': (string);
  'receiverAccountId': (string);
  'receiverAccountNumber': (string);
  'amount': (number);
  'currency': (string);
  'status': (_transaction_TransactionStatus__Output);
  'note': (string);
  'paymentCode': (string);
  'model': (string);
  'callNumber': (string);
  'createdAt': (string);
  'type': (_transaction_TransactionType__Output);
}

File: Backend/services/TransactionService/models/transaction.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type TransactionStatus string

const (
	StatusApproved TransactionStatus = "approved"
	StatusPending TransactionStatus = "pending"
	StatusDeclined TransactionStatus = "declined"
)

type Transaction struct {
	ID uuid.UUID `gorm:"type:uuid;primaryKey;default:gen_random_uuid()" json:"id"`
	SenderID uuid.UUID `gorm:"type:uuid;index;not null" json:"sender_id"`
	SenderAccountID uuid.UUID `gorm:"type:uuid;index;not null" json:"sender_account_id"`
	ReceiverID uuid.UUID `gorm:"type:uuid;index;not null" json:"receiver_id"`
	ReceiverAccountID uuid.UUID `gorm:"type:uuid;index;not null" json:"receiver_account_id"`
	TransactionStatus TransactionStatus `gorm:"type:varchar(50);not null;default:'pending'" json:"status"`
	CreatedAt time.Time `json:"created_at"`
	Amount float64 `gorm:"type:decimal(15,2);not null;default:0" json:"amount"`
	Currency          string         `gorm:"type:varchar(3);not null;default:'USD'" json:"currency"`
	Note string `gorm:"type:varchar(100);not null;default:''" json:"note"`
	PaymentCode string `gorm:"type:varchar(3);not null;default:''" json:"payment_code"`
	Model string `gorm:"type:varchar(2);not null;default:''" json:"model"`
	CallNumber string `gorm:"type:varchar(30);not null;default:''" json:"call_number"`
}